code/game/gamemodes/game_mode.dm

```diff
@@ -637,6 +637,16 @@
 		if(SSshuttle.emergency.is_hijacked())
 			SSticker.news_report = SHUTTLE_HIJACK
 
+/**
+  * Given a list of minds, returns TRUE if they completed all their objectives, FALSE otherwise
+  */
+/datum/game_mode/proc/didAntagsWin(list/antags)
+	for(var/datum/mind/m in antags)
+		for(var/datum/antagonist/a in m)
+			for(var/datum/objective/objective in a.objectives)
```

Comment:
All three loops dont need to be typed, make untyped loops and typecast it after
Example:
for(var/I in moblist)
   var/mob/M = I
instead of
for(var/mob/M in moblist)